app/controllers/regapps_controller.rb

```diff
@@ -5,14 +5,34 @@ def index
   end
 
   def show
+    @user = current_user
     @regapp = current_user.regapps.find(params[:id])
+    @regapp.user = current_user
```

Comment:
You shouldn't need to assign this here. If the Regapp is created, the user should be assigned. Also since you don't call save() after, no changes will be sent to the DB.